signer/rsapss/rsapss.go

```diff
@@ -79,24 +95,23 @@ func (s *RSAPSSSigner) Config() signer.Configuration {
 
 // SignData takes data, hashes it and returns a signed base64 encoded hash
 func (s *RSAPSSSigner) SignData(data []byte, options interface{}) (signer.Signature, error) {
-	h := sha1.New()
+	h := s.rsaPssHash.New()
 	h.Write(data)
 	digest := h.Sum(nil)
 	return s.SignHash(digest, options)
 }
 
 // SignHash takes an input hash and returns a signed base64 encoded hash
 func (s *RSAPSSSigner) SignHash(digest []byte, options interface{}) (signer.Signature, error) {
-	if len(digest) != 20 {
-		return nil, errors.Errorf("rsapss: refusing to sign input hash. Got length %d, expected 20.", len(digest))
-	}
-
-	opts := &rsa.PSSOptions{
-		SaltLength: rsa.PSSSaltLengthEqualsHash,
-		Hash:       crypto.SHA1,
-	}
-
-	sigBytes, err := s.key.(crypto.Signer).Sign(s.rng, digest, opts)
+	// sha1 is 20, sha256 is 32
+	if len(digest) != 20 && len(digest) != 32 {
```

Comment:
nit: using the pkg constants like sha1.Size would be clearer e.g. https://play.golang.org/p/_ncgf_SxFpT and let us drop the comment